src/lex.cpp

```diff
@@ -634,34 +657,55 @@ void lexer::parse_binary_number() {
   this->input_ = check_garbage_in_number_literal(input);
 }
 
-const char8* lexer::check_garbage_in_number_literal(const char8* input) {
+void lexer::parse_octal_number(bool strict) {
+  const char8* input = this->input_;
+
+  while (this->is_octal_digit(*input)) {
+    ++input;
+  }
+
   const char8* garbage_begin = input;
   for (;;) {
     switch (*input) {
+    /* TODO (👮🏾‍♀️ mswitch switch to just 8 and 9)*/
+    // decimal point should also be included
     QLJS_CASE_DECIMAL_DIGIT:
+      if (strict) {
+        ++input;
+        goto done_parsing_garbage;
+      } else {
+        this->input_ = input;
+        this->parse_number();
+        return;
+      }
     QLJS_CASE_IDENTIFIER_START:
-      input += 1;
-      break;
+    case u8'.':
+      ++input;
+      goto done_parsing_garbage;
       default:
         goto done_parsing_garbage;
     }
   }
+  /* note(👮🏾‍♀️)this is bad, because it won't parse the trailing
+   * garbage separate tokens correctly. for example, `0o57.2` should be two
+   * number tokens, while `0o5a` should be an "unknown" token and an "any" token
+   */
```

Comment:
This was back when I was narrow-mindedly focused on V8, instead of realizing the freedom we could take

```suggestion
```